### PR TITLE
Use a separate branch instead of gbp-balances.toml, create separate files

### LINKED ISSUE #23: Use empty branch for storing the data.
Instead of using master/workflow branch to store it create empty gbp_store/configurable empty branch with only the tally file.
Would keep the update commits out of main history.

### LINKED ISSUE #24: GBP theft on your repo!? It's more likely than you think.
Apparently when merging prs quickly github-action will not let the last action finish before starting the next one causing it to overwrite the last gbp change.

https://github.com/tgstation/tgstation/commit/bd40607e8978af71cc180857fcbff693485f035d
https://github.com/tgstation/tgstation/commit/bd40607e8978af71cc180857fcbff693485f035d

these are the two instances of this happening i could find.

### PR DESCRIPTION
**Pull request by Mothblocks**
Closes #23.
Addresses, but doesn't fix #24--if you merge two PRs in quick succession from the same person, you'll still meet the race condition.

- [ ] Remove all unused stuff

### CODE REVIEW

File: dist/index.js [hunk 1]
```diff
@@ -13149,11 +13149,10 @@ Object.defineProperty(exports, "__esModule", ({ value: true }));
 exports.merged = void 0;
 const core = __importStar(__nccwpck_require__(3923));
 const github = __importStar(__nccwpck_require__(5873));
-const toml = __importStar(__nccwpck_require__(8830));
 const isMaintainer_1 = __nccwpck_require__(7470);
 const points = __importStar(__nccwpck_require__(5474));
 function merged(configuration) {
-    var _a, _b, _c, _d, _e, _f;
+    var _a, _b, _c, _d, _e, _f, _g, _h, _j;
```
Comment (ZeWaka) [inline dist/index.js:13155]: you should be ashamed

File: dist/index.js [hunk 2]
```diff
@@ -13181,30 +13180,7 @@ function merged(configuration) {
...
-            .then((contents) => {
-            const data = contents.data;
-            return Array.isArray(data) ? undefined : data.sha;
-        })
-            .catch(() => {
-            // Most likely 404
-            return undefined;
-        });
-        yield octokit.repos.createOrUpdateFileContents(Object.assign(Object.assign({}, fileContentsParams), { message: `Updating GBP from PR #${pullRequest.number} [ci skip]`, content: Buffer.from(newOutput, "binary").toString("base64"), sha }));
+        yield points.writeBalanceOf(octokit, configuration.branch, (_e = (_d = github.context.payload.repository) === null || _d === void 0 ? void 0 : _d.owner) === null || _e === void 0 ? void 0 : _e.login, (_f = github.context.payload.repository) === null || _f === void 0 ? void 0 : _f.name, `Updating GBP from PR #${pullRequest.number} [ci skip]`, user.id, balance);
```
Comment (ZeWaka) [inline dist/index.js:13183]: what he fck

Comment (Mothblocks) [reply dist/index.js:13183]: its the compiled dist you dip, you need it for github actions because it doesnt support typescript

Comment (Mothblocks) [reply dist/index.js:13183]: this entire file is `ncc build src/index.ts -o dist`

Comment (ZeWaka) [reply dist/index.js:13183]: 😂

**Comment by AffectedArc07** [comment]
If the action runs in 5 seconds flat that should prevent race conditions, assuming the PR is being reviewed properly. I can’t think of being able to even merge 2 PRs in the space of 5 seconds unless you have them all lined up ready to go.

**Comment by Mothblocks** [comment]
It's happened to me before, lining up PRs while I wait for their CI to pass, but something else of note is that it might not be one person merging 2 PRs, but 2 people merging PRs at the same time.

**Comment by Mothblocks** [comment]
I'm closing this specific implementation, but separate branches WILL come! @stylemistake gave me a much better solution that completely eliminates the race conditions altogether (which this doesn't do, just obscures it) that I'm going to give a shot.

**Pull request closed without merging**